src/live_restore/live_restore_fs.c

```diff
@@ -799,7 +798,12 @@ __live_restore_fh_find_holes_in_dest_file(
     }
 
 err:
-    close(fd);
+    /*
+     * This should be wrapped in WT_SYSCALL but that will overwrite prior error codes. It's more
+     * important to catch errors reported earlier in this function than catching an error from
+     * closing the file.
+     */
+    WT_IGNORE_RET(close(fd));
```

Comment:
Would WT_TRET be better? It only overwrites ret if not set. With your changes, you may silence errors if everything was fine till now.

Reply:
Yeah fair point. `TRET` doesn't work because syscall's behave differently to normal WT calls, but I've added a new `WT_SYSCALL_TRET` macro for this.
I also realised I wasn't handling syscall errors properly on file open so good callout!

Reply:
Nice!